stack.py

```diff
@@ -168,9 +160,90 @@ def parse_program(code):
         yield Instr(op, args, prefix)
 
 
+def is_well_formed(parts):
+    r"""
+    Take an instruction split into a list of parts and check if it is well formed
+    Does not typecheck the instructions, only that they make syntaxtical sense
```

Comment:
Perhaps take a look at https://www.python.org/dev/peps/pep-0257/ for reference. A few docstring problems that you might as well fix while you're doing other things:
- There should be a one-line description of the function with whitespace following it. This is the "title." Your current first line looks perfect, but you need to leave a blank line after.
- You have no backslashes in here, so you don't need `r"""`, `"""` is fine.
- You should use proper punctuation in the body of the docstring.
- Perhaps don't leave the extra blank lines between your doctests, e.g.
```python
>>> is_well_formed(['@label'])
True
>>> is_well_formed(['@label', 'add'])
False
```
  would be fine. (Also, use `>>>`, not `>>`, then our test suite will run your examples as tests.)